pkg/sr/serde.go

```diff
@@ -134,6 +131,20 @@ func (s *Serde) SetDefaults(opts ...SerdeOpt) {
 	s.defaults = opts
 }
 
+// SetHeader configures which header should be used when encoding and decoding
+// values. If the header is set to nil it falls back to ConfluentHeader.
+func (s *Serde) SetHeader(header SerdeHeader) {
+	s.header = header
+}
+
+// Header returns the configured header.
+func (s *Serde) Header() SerdeHeader {
```

Comment:
I think this can be private

Reply:
Ugh, that would defeat the purpose of why I added this in the first place. I'd like to have access to the header and decode the ID manually (see my example in the PR description). Do you think there's a better way to achieve this? Another way would be to create the method `func (*Serde) DecodeID([]byte) (int, error)` which passes the call to the underlying `SerdeHeader`.

Reply:
Oh I see now. Would you be ok with `DecodeID`? That's a bit closer in spirit to the current API choices of the package, but I'd be ok with `Header()` if need be. There doesn't seem to be any purpose to exposing a getter for the header _besides_ decoding the ID of a slice -- I can't imagine a person will want to get the header to append an ID / I don't know what a use case for that could be. If you can think of a use case for Header besides ID decoding, I'm +1 to keeping the getter, otherwise I'm on the fence.